packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain.tsx

```diff
@@ -572,13 +599,26 @@ export function TransferPanelMain({
       return
     }
 
-    if (!ethBalance) {
+    // For custom fee token deposits, we can set the max amount, as the fees will be paid in ETH on L2
+    // For custom fee token withdrawals, they can be handled same as ETH withdrawals
+    if (nativeCurrency.isCustom && isDepositMode && customFeeTokenBalance) {
+      setAmount(
+        utils.formatUnits(customFeeTokenBalance, nativeCurrency.decimals)
+      )
+      return
+    }
+
+    const nativeCurrencyBalance = nativeCurrency.isCustom
+      ? customFeeTokenBalance
+      : ethBalance
```

Comment:
the way this is used makes me think that we should kill `ethBalance` in this function and only use `feeTokenBalance` as `customFeeTokenBalances` already handled the ETH case